src/main/java/picard/illumina/parser/PerTileFileUtil.java

```diff
@@ -22,9 +22,9 @@ public PerTileFileUtil(final String extension, final File base,
         super(true, extension, base, faker, lane, skipEmptyFiles);
         this.fileMap = getTiledFiles(base, matchPattern);
         if (!fileMap.isEmpty()) {
-            this.tiles = Collections.unmodifiableList(new ArrayList<Integer>(this.fileMap.keySet()));
+            this.tiles = new ArrayList<>(this.fileMap.keySet());
```

Comment:
Setting a field in a base class is not a great way to do this initialization. If it can't be passed to the base class constructor, maybe it should be represented by an abstract getter in the base class, and only stored in the subclasses.